the one that refactors the masthead

sturobson opened the pull request:
todo: 

- [ ] changelog.
- [ ] update documentation as needed.
- [ ] show other theme variants.


@khawkins98 note - you might have to go into `_preview--nogrid.njk` and change the urls for the css link and the JS script - so they're pointing to 

`<link rel="stylesheet" href="http://localhost:3003/vf-core/css/styles.css">`

`<script src="http://localhost:3003/vf-core/scripts/scripts.js"></script>`

if you're running `gulp dev` from inside `tools/vf-component-library` < that's how I've been doing it so far.

sturobson committed:
sneaks in hiding vscode workspaces

sturobson committed:
removes masthead search CSS

sturobson committed:
removes vf-masthead support as it's no longer needed

sturobson committed:
updates to masthead and variants

sturobson committed:
updates yaml/nunjuck relationship with variables

sturobson committed:
updates masthead variants

sturobson committed:
takes this back to normal

khawkins98 commented on components/vf-masthead/vf-masthead.config.yml:
```diff
@@ -12,23 +12,31 @@ variants:
-      mastheadImage: "../../assets/vf-masthead/assets/group-bg_2d4155.png"
-      hasTitleBlock: vf-masthead--with-title-block
+      mastheadImageClass: true
+      mastheadImage: "http://localhost:3003/vf-core/assets/vf-masthead/assets/group-bg_2d4155.png"
```
I know this is in progress, but if we put the images on https://www.embl.org/files we can make use of the CDN and not have to worry about paths or storing the demonstration images in version control.

khawkins98 commented:
Having a look at the JS, I think there's a couple ways to make it work "cleaner".

khawkins98 committed:
Refactore: vf-masthead-js

Optimastions for #1195, this:

- lints the JS
- allows use of `style=" --vf-masthead__bg-image: url(../../assets/vf-masthead/assets/group-bg_2d4155.png);`
- fixes the JS to remove escaped syntax

khawkins98 committed:
Update vf-masthead.js

sturobson committed:
updates masthead JS, image, njk variables

sturobson committed:
cupdates theming variable for title-block--primary

sturobson committed:
makes words break on smaller devices

sturobson committed:
more updates to masthead, cleaner code, improved docs

sturobson committed:
cha-cha-cha-changelog
